pygount/write.py

```diff
@@ -128,154 +131,51 @@ def close(self):
-        separator_line = "  ".join(
-            [
-                "-" * width
-                for width in (
-                    max_language_width,
-                    max_file_count_width,
-                    percentage_width,
-                    max_code_width,
-                    percentage_width,
-                    max_documentation_width,
-                    percentage_width,
-                )
-            ]
-        )
-        self._target_stream.write(separator_line + os.linesep)
-        language_line_template = (
-            "{0:{max_language_width}s}  "
-            "{1:>{max_file_count_width}d}  "
-            "{2:>{percentage_width}.0{digits_after_dot}f}  "
-            "{3:>{max_code_width}d}  "
-            "{4:>{percentage_width}.0{digits_after_dot}f}  "
-            "{5:>{max_documentation_width}d}  "
-            "{6:>{percentage_width}.0{digits_after_dot}f}"
-        )
-        for language_summary in sorted(self.project_summary.language_to_language_summary_map.values(), reverse=True):
-            code_count = language_summary.code_count
-            code_percentage = (
-                code_count / self.project_summary.total_code_count * 100
-                if self.project_summary.total_code_count != 0
-                else 0.0
-            )
-            file_count = language_summary.file_count
-            assert (
-                self.project_summary.total_file_count != 0
-            ), "if there is at least 1 language summary, there must be a file count too"
-            file_percentage = file_count / self.project_summary.total_file_count * 100
-            documentation_percentage = (
-                language_summary.documentation_count / self.project_summary.total_documentation_count * 100
-                if self.project_summary.total_documentation_count != 0
-                else 0.0
-            )
-            line_to_write = language_line_template.format(
+        table = Table()
+
+        for col in self._TEXT_COLUMNS:
+            table.add_column(col)
+        for col in self._DIGIT_COLUMNS:
+            table.add_column(col, justify="right")
+        language_summaries = sorted(self.project_summary.language_to_language_summary_map.values(), reverse=True)
+
+        for index, language_summary in enumerate(language_summaries, start=1):
+            table.add_row(
                 language_summary.language,
-                file_count,
-                file_percentage,
-                code_count,
-                code_percentage,
-                language_summary.documentation_count,
-                documentation_percentage,
-                digits_after_dot=digits_after_dot,
-                max_code_width=max_code_width,
-                max_documentation_width=max_documentation_width,
-                max_file_count_width=max_file_count_width,
-                max_language_width=max_language_width,
-                percentage_width=percentage_width,
+                *map(
```

Comment:
Would a more pythonic list comprehension work here, too? AFAIK the `map` function is kind of an relic from the time when there was no list comprehension.

Something like:
```python
[str(value) for value in (
    language_summary.file_count,
    language_summary.empty_count,
    language_summary.documentation_count,
    language_summary.code_count,
)]
```

Reply:
Actually, map is not exactly a relic. It's just another method of doing such things in functional programming. Also, when you're not using lambdas, map is roughly 20%-40% faster than listcomp so I use it in all simple cases. However, if you prefer listcomps for such cases, we can use it -- the table creation is not necessary to optimize anyways.

Reply:
I found [the old discussion](https://www.artima.com/weblogs/viewpost.jsp?thread=98196) where Guido was thinking about removing `map()`, but it managed to survive this. But I agree with his point that list comprehension is easier to read. So if you don't mind I'd prefer it over `map()` here.

Good to know that there is a performance difference, thanks for enlightening me.

Reply:
Cleaned up.